### PR TITLE
Remove the back button on secure conversations chat transcript

### PR DESCRIPTION
**Pull request by gersonnoboa**
The current logic correctly identifies when to show the secure conversations title or the chat title. This logic is encapsulated into its own function so that it can be used to determine back button behavior. The whole props for the header are regenerated every time the new renderProps method (previously renderTitle) is called, so that it correctly identifies the different chat models and its actions.

MOB-1888

### CODE REVIEW

File: GliaWidgets/Sources/ViewModel/Chat/ChatViewModel.swift [hunk 1]
```diff
@@ -296,6 +266,38 @@ class ChatViewModel: EngagementViewModel, ViewModel {
     }
 }
 
+extension ChatViewModel {
```
Comment (gersonnoboa) [inline GliaWidgets/Sources/ViewModel/Chat/ChatViewModel.swift:269]: This change is only because SwiftLint was complaining

**Review by igorkravchenko — APPROVED** [review]
Looks good to me.

**Review by EgorovEI — APPROVED** [review]

**Review by yurii-glia — APPROVED** [review]

**Comment by gersonnoboa** [comment]
!merge

**Comment by gersonnoboa** [comment]
!squash

**Comment by gersonnoboa** [comment]
!merge

**Pull request merged**